File: src/Bank.java
import java.util.ArrayList;
import java.util.Iterator;
import java.util.Scanner;

public class Bank {
    // Array + Variables globales
    static Scanner scanner = new Scanner(System.in);
    public static ArrayList<BankAccount> accounts = new ArrayList<>();
    public static void seeAccounts(){
        int numId=1;
        Iterator<BankAccount> bankIterator = accounts.iterator();
        System.out.println("Lista de cuentas\n");
        System.out.println("Numero de cuenta\tTipo de cuenta\tSaldo en cuenta");
        while (bankIterator.hasNext()){
            BankAccount bank = bankIterator.next();
            System.out.println(numId+" - "+bank.getAccountNumber() +"\t"+ bank.getType() +"\t"+ bank.getAmount());
            numId++;
        }
        System.out.println("");
    }
    public static void seeAccount(BankAccount account){
        System.out.println("\nNumero de cuenta: "+account.getAccountNumber());
        System.out.println("Tipo de cuenta: "+account.getType());
        System.out.println("Saldo en cuenta: "+account.getAmount());
    }

    // Setters personalizados
    private static char setType(){
        char type = scanner.next().toUpperCase().charAt(0);;
        if (type != 'A' && type != 'B' && type != 'C') {
            System.out.println("Tipo de cuenta no válido");
            System.out.print("Tipo de cuenta: ");
            type = setType();
        }
        return type;
    }
    private static long setNumberAccount(){
        long number = scanner.nextLong();
        if (number <= 0){
            System.out.println("Numero invalido");
            System.out.print("Numero de cuenta: ");
            number = setNumberAccount();
        }
        return number;
    }
    public static void addAccount(){
        int numEmployee;
        long accountNumber;
        char type;
        System.out.print("Ingrese numero de empleado (Visible en opción 4 - Ingrese 0 para cancelar): ");
        numEmployee = scanner.nextInt()-1;
        if (numEmployee >= 0){
            System.out.println("Ingrese la información de la nueva cuenta");
            System.out.print("Numero de cuenta: ");
            accountNumber = setNumberAccount();
            System.out.print("Tipo de cuenta: ");
            type = setType();
            BankAccount account = new BankAccount(accountNumber, type);
            Employee.employeeArray.get(numEmployee).getAccountsArray().add(account);
            Bank.accounts.add(account);
            System.out.println("Cuenta agregada");
        } else {
            System.out.println("Operacion cancelada");
        }
    }

    // Metodos extras
    public static void createEmployee(){
        String name, lastName;
        int aux;
        System.out.println("¿Desea crear un usuario con cuenta?\n1 - Sí\n2 - No");
        aux = scanner.nextInt();
        if (aux == 2){
            System.out.print("Nombre: ");
            name = scanner.next();
            System.out.print("Apellido: ");
            lastName = scanner.next();
            Employee employee = new Employee(name, lastName);
            Employee.employeeArray.add(employee);
            System.out.println("Empleado agregado");
            seeEmployee(Employee.employeeArray.get(Employee.employeeArray.size()-1));
        }
        else if (aux == 1){
            createEmployeeAndAcount();
            seeEmployee(Employee.employeeArray.get(Employee.employeeArray.size()-1));
        }
        else {
            System.out.println("Opcion inexistente");
        }
    }
    public static void createEmployeeAndAcount(){
        String name, lastName;
        long account;
        char type;
        try{
            System.out.print("Nombre: ");
            name = scanner.next();
            System.out.print("Apellido: ");
            lastName = scanner.next();
            System.out.print("Numero de cuenta: ");
            account = setNumberAccount();
            System.out.print("Tipo de cuenta: ");
            type = setType();
            Employee employee = new Employee(name, lastName, account, type);
            Employee.employeeArray.add(employee);
            System.out.println("Empleado agregado");
        } catch (Exception e) {
            e.printStackTrace();
            System.out.println("Error en el ingreso de los datos");
        }
    }

    public static void seeEmployees(){
        int numId=1;
        Iterator<Employee> employeeIterator = Employee.employeeArray.iterator();
        System.out.println("Lista de empleados\n");
        while (employeeIterator.hasNext()){
            Employee employee = employeeIterator.next();
            if (employee.getAccountsArray().size() == 1) {
                System.out.println(numId+" - Empleado: "+employee.getName()+"\tNumero de cuenta: "+employee.getAccount().getAccountNumber()+"\tSaldo: "+employee.getAccount().getAmount()+"\tTipo de cuenta: "+employee.getAccount().getType());
            } else {
                System.out.println(numId+" - Empleado: "+employee.getName()+"\tCantidad de cuentas: "+employee.getAccountsArray().size());
            }
            numId++;
        }
        System.out.println("");
    }
    public static void seeEmployee(Employee employee){
        System.out.println("\nNombre: "+employee.getName());
        System.out.println("Apellido: "+employee.getLastName());
        seeEmployeeAcounts(employee);
    }
    private static void seeEmployeeAcounts(Employee employee){
        for (int i = 0; i<employee.getAccountsArray().size() ;i++){
            System.out.println("\nCuenta " + (i+1));
            System.out.println("Numero de cuenta: " + employee.getAccountsArray().get(i).getAccountNumber());
            System.out.println("Tipo de cuenta: "+ employee.getAccountsArray().get(i).getType());
            System.out.println("Saldo en cuenta: "+ employee.getAccountsArray().get(i).getAmount());
        }
    }

}

File: src/Bank.py
from BankAccount import BankAccount
from Employee import Employee
class Bank:
    accounts = []
    
    def see_accounts():
        num_id=1
        print("\nLista de cuentas")
        print("Numero de cuenta    Tipo de cuenta       Saldo en cuenta")
        for account in Bank.accounts:
            print(f"{num_id} - Cuenta: {account.get_account_number()}     Tipo: {account.get_type()}      Monto: {account.get_amount()}")
            num_id = num_id+1
        print("")
    
    def see_account():
        print("Ingrese numero de cuenta (Visible en opción 6 - Ingrese 0 para cancelar)")
        aux = int(input())
        if(aux > 0):
            aux = aux -1
        print(f"Numero de cuenta: {Bank.accounts[aux]}")
        print(f"Tipo de cuenta: {Bank.accounts[aux]}")
        print(f"Saldo en cuenta: {Bank.accounts[aux]}")

    def set_type():
        type = input()
        if type != 'A' and type != 'B' and type != 'C':
            print("Tipo de cuenta no valido")
            print("tipo de cuenta: ")
            type = Bank.set_type()
        return type

    def set_number_account():
        number = int(input())
        if number<=0:
            print("Numero invalido")
            print("Numero de cuenta: ")
            number = Bank.set_number_account()
        return number
    
    def add_account():
        print("\nIngrese numero de empleado (Visible en opcion 4 - Ingrese 0 para cancelar): ")
        num_employee = int(input())-1
        if num_employee >= 0:
            print("Ingrese la informacion de la nueva cuenta")
            print("Numero de cuenta")
            account_number = Bank.set_number_account()
            print("Tipo de cuenta: ")
            type = Bank.set_type()
            account = BankAccount(account_number, type)
            Employee.employee_array[num_employee].get_accounts_array().append(account)
            Bank.accounts.append(account)
            print("Cuenta agregada")
        else:
            print("Operacion cancelada")


    def create_employee():
        print("Nombre: ")
        name = input()
        print("Apellido: ")
        lastName = input()
        print("Numero de cuenta: ")
        account = Bank.set_number_account()
        print("Tipo de cuenta: ")
        type = Bank.set_type()
        employee = Employee(name, lastName, account, type)
        Employee.employee_array.append(employee)
        Bank.accounts.append(account)
        print("Empleado agregado")
                
        

    def see_employees():
        num_id=1;
        print(f"\nLista de empleados")
        for employee in Employee.employee_array:
            if (len(employee.accounts_array) == 1): 
                print(f"{num_id}- Empleado: {employee.get_name()}Numero de cuenta: {employee.get_account().get_account_number()}Saldo: {employee.get_account().get_amount()}Tipo de cuenta: {employee.get_account().get_type()}")
            
            else :
                print(f"{num_id} - Empleado: {employee.get_name()}        Cantidad de cuentas: {len(employee.get_accounts_array())}")
            
            num_id = num_id+1
        
        print(f"")
    
    def see_employee(employee):
        print(f"\nNombre: {employee.get_name()}")
        print(f"Apellido: {employee.get_last_name()}")
        Bank.see_employee_acounts(employee)
    
    def see_employee_acounts(employee):
        for i in range(len(employee.accounts_array())):
            y = i+1
            print(f"Cuenta {y}")
            print(f"Numero de cuenta:   {employee.accounts_array[i].get_account_number()}")
            print(f"Tipo de cuenta:  {employee.accounts_array[i].get_type()}")
            print(f"Saldo en cuenta:  {employee.accounts_array[i].get_amount()}")
        
